rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/connection/NotificationAndIndicationManager.java

```diff
@@ -138,17 +141,54 @@ public void run() {
     }
 
     @NonNull
-    private static CompletableTransformer setupModeTransformer(final DescriptorWriter descriptorWriter,
-                                                               final BluetoothGattCharacteristic characteristic,
-                                                               final byte[] value,
-                                                               final NotificationSetupMode mode) {
+    private static ObservableTransformer<Observable<byte[]>, Observable<byte[]>> setupModeTransformer(
+            final DescriptorWriter descriptorWriter,
+            final BluetoothGattCharacteristic characteristic,
+            final byte[] value,
+            final NotificationSetupMode mode
+    ) {
+        return new ObservableTransformer<Observable<byte[]>, Observable<byte[]>>() {
+            @Override
+            public ObservableSource<Observable<byte[]>> apply(final Observable<Observable<byte[]>> upstream) {
+                switch (mode) {
+
+                    case COMPAT:
+                        return upstream;
+                    case QUICK_SETUP:
+                        final Completable publishedWriteCCCDesc = writeClientCharacteristicConfig(characteristic, descriptorWriter, value)
+                                .toObservable()
+                                .cache()
```

Comment:
Is it safe? If you cache you give up the possibility of disposing of the observable.

Reply:
A good question. Writing CCC descriptor is a one time operation so it should end quickly. This question inspired me to write another test to check if writing the CCC will not happen after the parent `Observable<Observable<byte[]>>` is disposed